code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_spiders.dm

```diff
@@ -0,0 +1,328 @@
+		CheckFaction()
+	else if(istype(target, /obj/effect/spider/cocoon))
+		to_chat(src, "Destroying our own cocoons would not help us.")
+	else if(istype(target, /obj/machinery/door/firedoor))
+		var/obj/machinery/door/firedoor/F = target
+		if(F.density)
+			if(F.blocked)
+				to_chat(src, "The fire door is welded shut.")
+			else
+				visible_message("<span class='danger'>\The [src] pries open the firedoor!</span>")
+				F.open()
+		else
+			to_chat(src, "Closing fire doors does not help.")
+	else if(istype(target, /obj/machinery/door/airlock))
+		var/obj/machinery/door/airlock/A = target
+		if(A.density)
+			try_open_airlock(A)
+	else if(isliving(target))
+		var/mob/living/G = target
+		if(issilicon(G))
+			G.attack_animal(src)
+			return
+		else if(G.reagents && (iscarbon(G)))
+			var/can_poison = 1
+			if(istype(G, /mob/living/carbon/human/))
+				var/mob/living/carbon/human/H = G
+				if(H.dna)
+					if(!(H.species.reagent_tag & PROCESS_ORG) || (H.species.flags & NO_POISON))
+						can_poison = 0
+			spider_specialattack(G,can_poison)
+		else
+			G.attack_animal(src)
+	else
+		target.attack_animal(src)
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/spider_specialattack(var/mob/living/carbon/human/L, var/poisonable)
+	L.attack_animal(src)
+
+
+
+
+
+
+// --------------------------------------------------------------------------------
+// --------------------- TERROR SPIDERS: PROC OVERRIDES ---------------------------
+// --------------------------------------------------------------------------------
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/examine(mob/user)
+	..()
+	var/msg = ""
+	if(stat == DEAD)
+		msg += "<span class='deadsay'>It appears to be dead.</span>\n"
+	else
+		if(key)
+			msg += "<BR><span class='warning'>Its eyes regard you with a curious intelligence.</span>"
+		if(health > (maxHealth*0.95))
+			msg += "<BR><span class='notice'>It is in excellent health.</span>"
+		else if(health > (maxHealth*0.75))
+			msg += "<BR><span class='notice'>It has a few injuries.</span>"
+		else if(health > (maxHealth*0.55))
+			msg += "<BR><span class='warning'>It has many injuries.</span>"
+		else if(health > (maxHealth*0.25))
+			msg += "<BR><span class='warning'>It is barely clinging on to life!</span>"
+		else if(health < maxHealth && regen_points > regen_points_per_kill)
+			msg += "<BR><span class='notice'>It appears to be regenerating quickly</span>"
+		if(killcount == 1)
+			msg += "<BR><span class='warning'>It is soaked in the blood of its prey.</span>"
+		if(killcount > 1)
+			msg += "<BR><span class='warning'>It is soaked with the blood of " + num2text(killcount) + " prey it has killed.</span>"
+	to_chat(usr,msg)
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/New()
+	..()
+	ts_spiderlist += src
+	if(type == /mob/living/simple_animal/hostile/poison/terror_spider)
+		message_admins("[src] spawned in [get_area(src)] - a subtype should have been spawned instead.")
+		qdel(src)
```

Comment:
You might also want `log_runtime(EXCEPTION("Base type of the terror spiders created at [atom_loc_line(src)]"))` - this will add an entry to our error logger about the issue.